src/rank_llm/rerank/listwise/listwise_inference_handler.py

```diff
@@ -1,53 +1,161 @@
-from typing import Any, Dict, List, Optional, Tuple
+import re
+from typing import Any, Dict, List, Tuple
+
+from ftfy import fix_text
 
 from rank_llm.data import Result
 from rank_llm.rerank.inference_handler import BaseInferenceHandler
 
+ALPH_START_IDX = ord("A") - 1
+
 
-# TODO(issue #237): Need to modify functions for this class
 class ListwiseInferenceHandler(BaseInferenceHandler):
     def __init__(self, template: Dict[str, str]):
         super().__init__(template)
 
+        self._validate_template(self.template)
+
     def _validate_template(self, template: Dict[str, str]):
-        pass
+        required_template_keys = {
+            "method": ["listwise"],
+            "body": ["{rank}", "{candidate}"],
+        }
+
+        allowed_template_keys = {
+            "system_message": [],
+            "prefix": ["{num}", "{query}"],
+            "suffix": ["{num}", "{query}"],
+        }
+
+        # Validate the required keys
+        missing_template_keys = [
+            key for key in required_template_keys if key not in template
+        ]
+        if missing_template_keys:
+            raise ValueError(f"Missing required template keys: {missing_template_keys}")
+
+        # Validate the method value
+        if template["method"] != "listwise":
+            raise ValueError(
+                f"Incorrect method type, expected \"listwise\", got {template['method']}"
+            )
+
+        # Validate the rest of the template keys
+        for template_key, template_value in template.items():
+            if (
+                template_key not in required_template_keys
+                and template_key not in allowed_template_keys
+            ):
+                raise ValueError(f"Unknown template key: {template_key}")
+            if template_key in required_template_keys:
+                for keyword in required_template_keys[template_key]:
+                    if keyword not in template_value:
```

Comment:
template value here is a formatted string with {}. the keyword existing in the str is not enough it should be between {}. use formatter.parse(template_value) to find its keys.

This way you can also add a strict mode that when enabled raises value error for unsupported keys in the str templates. Otherwise just log a warning